pdr_backend/lake/duckdb_data_store.py

```diff
@@ -243,34 +288,79 @@ def query_data(self, query: str) -> Optional[pl.DataFrame]:
             raise e
 
     @enforce_types
-    def move_table_data(self, temp_table, permanent_table):
+    def get_query_drop_records_from_table_by_id(
+        self, drop_table_name: str, ref_table_name: str
+    ):
         """
-        Move the table data from the temporary storage to the permanent storage.
+        Builds the query string to perform the operation and returns it
         @arguments:
-            temp_table - The temporary table object
-            permanent_table_name - The name of the permanent table.
+            drop_table_name - The table to drop records from.
+            ref_table_name - The table to reference for the IDs to drop.
+        @example:
+            get_query_drop_records_from_table_by_id("bronze_pdr_slots", "update_pdr_slots")
+        """
+
+        # Return the query string
+        return f"""
+        DELETE FROM {drop_table_name}
+        WHERE ID IN (
+            SELECT DISTINCT ID 
+            FROM {ref_table_name}
+        );
+        """
+
+    @enforce_types
+    def drop_records_from_table_by_id(self, drop_table_name: str, ref_table_name: str):
+        """
+        Drop the records from the table by the provided IDs.
+        @arguments:
+            table_name - A unique name for the table.
+            ids - The list of IDs to drop.
+        @example:
+            drop_records_from_table_by_id("bronze_pdr_slots", "update_pdr_slots")
+        """
+        query = self.get_query_drop_records_from_table_by_id(
+            drop_table_name, ref_table_name
+        )
+        self.execute_sql(query)
+
+    @enforce_types
+    def get_query_move_table_data(self, from_table, to_table):
```

Comment:
let's use "source" and "destination"/"dest" instead of to/from.

Reply:
why is that?

"from_" and "to_" is common nomenclature throughout the whole engine... "source_" and "destination_" isn;t.

not only are they longer, but not common practice in the rest of the code base